### PR TITLE
Allow configure Controller Manager compute resources

### PR DESCRIPTION
**Pull request by jorge07**

### CODE REVIEW

File: core/controlplane/config/templates/cluster.yaml [hunk 1]
```diff
@@ -1094,6 +1094,14 @@ worker:
 #kubernetes:
 #  networking:
 #    selfHosting:
+#  controllerManager:
+#    resources:
+#      requests:
+#        cpu: 200m
```
Comment (c-knowles) [inline core/controlplane/config/templates/cluster.yaml]: Usually the values here are the defaults. Are there any recommended defaults from upstream?

Comment (jorge07) [reply core/controlplane/config/templates/cluster.yaml]: This is the default value righ know for cpu. That I added for limits are based on our production cluster requirements. If you know where I can find any recommendation for this, please, share

File: core/controlplane/config/templates/cloud-config-controller [hunk 2]
```diff
@@ -3323,7 +3323,19 @@ write_files:
           {{end}}
           resources:
             requests:
-              cpu: 200m
+              cpu: {{ if .Kubernetes.ControllerManager.ComputeResources.Requests.Cpu }}{{ .Kubernetes.ControllerManager.ComputeResources.Requests.Cpu }}{{ else }}200m{{ end }}
```
Comment (mumoshu) [inline core/controlplane/config/templates/cloud-config-controller]: Nit: We'd better define the default of `200m` in go and `cluster.yaml` template, similarly to other settings.

**Comment by mumoshu** [comment]
LGTM. Thank you so much for your contribution @jorge07!

I'd appreciate it if you could update the defaults once we find any in upstream

File: core/controlplane/config/templates/cluster.yaml (hunk 1 above)
Comment (mumoshu) [reply core/controlplane/config/templates/cluster.yaml]: Let's continue the discussion in #1254

**Pull request merged**